projects/core/src/test/resources/test-rom/spec/apis/textutils_spec.lua

```diff
@@ -174,6 +176,17 @@ describe("The textutils library", function()
             expect(textutils.serializeJSON({ 5, "test", nil, nil, textutils.json_null })):eq('[5,"test",null,null,null]')
             expect(textutils.serializeJSON({ nil, nil, nil, nil, "text" })):eq('[null,null,null,null,"text"]')
         end)
+
+        it("serializes NBT style", function()
+            expect(textutils.serializeJSON({ test = 2 }, { nbt_style = true })):eq('{test:2}')
+            expect(textutils.serializeJSON({ test = 2 }, true)):eq('{test:2}') -- old style
+        end)
+
+        it("serializes Unicode strings", function()
+            expect(textutils.serializeJSON("\xe3\x81\x93\xe3\x82\x93\xe3\x81\xab\xe3\x81\xa1\xe3\x81\xaf", { unicode_strings = true })):eq('"\\u3053\\u3093\\u306B\\u3061\\u306F"')
+            expect(textutils.serializeJSON("\xf0\x9f\x98\xaf", { unicode_strings = true })):eq('"\\uD83D\\uDE2F"')
+            expect(textutils.serializeJSON("\\\"\xc3\xbf\n\"", { unicode_strings = true })):eq('"\\\\\\"\\u00FF\\n\\""')
```

Comment:
This might be a little cleaner if we use unicode escapes and long strings, as we can remove a bunch of escapes:

```suggestion
            expect(textutils.serializeJSON("\u{3053}\u{3093}\u{306B}\u{3061}\u{306F}", { unicode_strings = true })):eq([["\u3053\u3093\u306B\u3061\u306F"]])
            expect(textutils.serializeJSON("\u{1f62f}", { unicode_strings = true })):eq([["\uD83D\uDE2F"]])
            expect(textutils.serializeJSON("\\\"\u{00ff}\n\"", { unicode_strings = true })):eq('"\\\\\\"\\u00FF\\n\\""')
```